### PR TITLE
Remove Polymer

### LINKED ISSUE #213: how to reset graph to initial state?
(forgive my psuedocode)
Is there a way to set a the-graph instance back to the initial state?

We have this structure :

```
<our-editor>
  <the-editor>
    graph
    nofloGraph
    <the-graph graph={{nofloGraph}}>
```

We have a button that is supposed to reset our editor back to its initial state, and we do that by :

```
<our-editor>.reset()
  <the-editor>.graph = {};
```

However, when reset() as above, and then load another graph, it is clear that there is still some 'state' left over from the previous graph - if a node in the new graph has the same key name as one in the old graph, then the new one inherits the ports from the old one. This is only obvious momentarily, so, for example, if the old node has an input port, and the new one has an output port, then the user can see that the 'new' node is added with an input port, but that then is removed and replaced with an output port.

The end result is correct, but we'd like to be sure that the 'reset()' is actually thorough. Is there any way to avoid this glitch?

### LINKED ISSUE #222: Get rid of Polymer
Hi,

I wonder if anyone can give me some information regarding no-flo's plans for moving to Polymer 1.0 from 0.5.X?

Thanks,

Max.

### LINKED ISSUE #290: /deep/ combinator is deprecated: refactor styles
needs research

### LINKED ISSUE #344: Upgrade to Polymer 2
I have a branch where I've successfully upgraded to Polymer 2 :)

[the-graph#polymer2](https://github.com/kristianmandrup/the-graph/tree/polymer2)

Would anyone be interested in helping out to clean it up further and also migrate to class based latest version of React? 

Cheers!

### PR DESCRIPTION
**Pull request by jonnor**
Fixes #222 
Fixes #213
Closes #290 
Closes #344

The full demo has some features missing compared to before. There are APIs on the React element that can be used as a stopgap, but this is hacky and should go away. However these APIs are still in use by the Polymer elements in noflo-ui. This affects:

Easyfix
* [x] Navigation thumbnail shows wrong editor viewport. Not wired up in demo

Needs more API
* Node and edge selection. React element need to respect props, demo needs to listen to callbacks
* Node errors. React element needs to respect props, demo updated to use them
* Some keyboard shortcuts might fail, as they expect API from previous Polymer element. React element should have a callback in props instead of hardcoding

Harder to fix
* Focus node. React element must support props for controling graph view (pan/zoom)
* Fit-to-view. Same as above
* Pan editor via dragging navigation. Same as above 

Not planning to fix
* Autolayout. This only impacts `graph` state, which is outside React elements. Applications should use klayjs-noflo directly, if wanted
* Clipboard handling. Apps can do this themselves, only impacts `graph` state

### CODE REVIEW

**Comment by jonnor** [comment]
Tested briefly in noflo-ui, seems fine

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,6 @@ node_js:
 - node
 before_script:
 - npm install -g grunt-cli
-- bower install
```
Comment (bergie) [inline .travis.yml]: yay!

**Comment by jonnor** [comment]
@bergie there was an implicit bind of the getMenuDef callback before, that needs an explicit bind in noflo-ui now. I got a patch for it. Otherwise no changes needed

**Comment by forresto** [comment]
Amazing... cut the Gordian knot!

**Comment by kristianmandrup** [comment]
Sweet :)

**Pull request merged**